### PR TITLE
skeleton: reconstruct pal init api v1 and v3

### PR DESCRIPTION
**Pull request by YangLiang3**
Signed-off-by: Liang Yang <liang3.yang@intel.com>

### CODE REVIEW

File: rune/libenclave/internal/runtime/pal/api_linux_v3.go [hunk 1]
```diff
@@ -3,6 +3,24 @@ package enclave_runtime_pal // import "github.com/inclavare-containers/rune/libe
 /*
 #include <stdlib.h>
 #include <errno.h>
+#include "skeleton/liberpal-skeleton.h"
+
+static int palGetVersion(void *sym)
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/api_linux_v3.go:8]: I believe you don't need to re-implement this APU in v3.

File: rune/libenclave/internal/runtime/pal/api_linux_v3.go [hunk 2]
```diff
@@ -16,12 +34,43 @@ import "C"
...
 	"fmt"
 	"github.com/inclavare-containers/rune/libenclave/intelsgx"
+	"github.com/sirupsen/logrus"
 	"unsafe"
 )
 
 type enclaveRuntimePalApiV3 struct {
 }
 
+func (pal *enclaveRuntimePalApiV3) get_version() uint32 {
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/api_linux_v3.go:44]: You don't need to re-implement this function in v3.

File: rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c [hunk 3]
```diff
@@ -456,6 +442,25 @@ int __pal_init(pal_attr_t *attr)
 	if (!encl_build(&secs, bin, bin_size, &sigstruct, &token))
 		return -EINVAL;
 
+	return 0;
+}
+
+/* *INDENT-OFF* */
+int __pal_init_v1(pal_attr_v1_t *attr)
+{
+	int ret;
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c:451]: Add a newline plz.

**Review by jiazhang0 — CHANGES_REQUESTED** [review]

File: rune/libenclave/internal/runtime/pal/api_linux_v3.go (hunk 1 above)
Comment (hustliyilin) [reply rune/libenclave/internal/runtime/pal/api_linux_v3.go:8]: You Can refer to API v2 which doesn't re-implement this API.

File: rune/libenclave/internal/runtime/pal/api_linux_v3.go (hunk 2 above)
Comment (hustliyilin) [reply rune/libenclave/internal/runtime/pal/api_linux_v3.go:44]: You Can refer to API v2 which doesn't re-implement this API.

File: rune/libenclave/internal/runtime/pal/api_linux_v3.go (hunk 1 above)
Comment (YangLiang3) [reply rune/libenclave/internal/runtime/pal/api_linux_v3.go:8]: Thank for the suggestion.
it seems that there are some confusion on pal layer. 
But I will change as you said.

**Pull request closed without merging**